### ISSUE TITLE
Couldn't start client D Language

### ISSUE THREAD

**Issue by RUSshy** [LaurentTreguier/vscode-dls#1]
Hi,

I tried install the extention on vscode but when i load my project folder i get the error:

```
[Error - 4:45:53 PM] Starting client failed
Error: Unsupported server configuration {
    "command": ""
}
	at _getServerWorkingDir.then.serverWorkingDir (C:\Users\rus\.vscode\extensions\LaurentTreguier.vscode-dls-1.4.1\node_modules\vscode-languageclient\lib\main.js:355:35)
	at <anonymous>
```

This is a new empty project i just created with dub

And if i added the dls in my dub and i run ``dub`` command it downloads everything but then:

```
λ dub
Running pre-generate commands for dscanner...
Running pre-generate commands for dls...
The system cannot find the path specified.
Command failed with exit code 1
```

**Comment by LaurentTreguier** [comment]
I'm currently in the process of setting up my PC and installing everything, so I'll be able to really look into this only tomorrow at best...
What happens if you run `dub run dls:bootstrap -- -m=download -c` ?

**Comment by LaurentTreguier** [comment]
Apparently Windows is too strict to let normal users make symbolic links... Even though on my older machine it worked perfectly. To quick-fix the problem, you can launch an admin command prompt, go to `C:\Users\rus\AppData\Local\dub\packages\.bin`, run `mklink dls.exe dls-v0.5.0\dls.exe` and set then the extension's `d.dlsPath` setting to `C:\Users\rus\AppData\Local\dub\packages\.bin\dls.exe`.
This should be sufficient to make it work until I find a way to work around Windows' shortcomings.

**Comment by LaurentTreguier** [comment]
Ah, it looks like enabling Windows' Delelopper Mode on Windows 10 allows unprivileged users to create symlinks. If you are running Windows 10, you can either use the previous solution, or enable the Developper Mode in the settings app, under the 'Updates and security' category.

**Comment by RUSshy** [comment]
I tried enabled developper mode, but it failed

I tried the link method but there is no folder .bin  and if i try to create the folder manually windows tell me it is not valid name because of the dot

**Comment by RUSshy** [comment]
And if i try run your command i get this

![image](https://user-images.githubusercontent.com/18348637/40579160-b9b7c57a-6121-11e8-8331-f728598431d9.png)

**Comment by LaurentTreguier** [comment]
Ok, I think it's safe to say that I'll need quite some time to fix it on Windows. have you tried code-d for VSCode ?

**Comment by RUSshy** [comment]
I cd to dls folder to build it with, and it fails at the link part

```
Linking...
c:\d\dmd2\windows\bin\dmd.exe -of.dub\build\application-debug-windows-x86-dmd_2080-13BEFD4853E9E0CD533ADC8D74C0166A\dls.exe .dub\build\application-debug-windows-x86-dmd_2080-13BEFD4853E9E0CD533ADC8D74C0166A\dls.obj ..\..\dcd-0.9.8\dcd\.dub\build\library-debug-windows-x86-dmd_2080-22A618EF6FB7C5AAC2581D12A707DA86\dcd.lib ..\..\msgpack-d-1.0.0-beta.6\msgpack-d\.dub\build\library-debug-windows-x86-dmd_2080-B19D9263F7FE78E5DC3920320559D15B\msgpack-d.lib ..\..\dfmt-0.7.0\dfmt\.dub\build\library-debug-windows-x86-dmd_2080-B09D539D75F8350874778994ED1EA705\dfmt.lib bootstrap\.dub\build\library-debug-windows-x86-dmd_2080-9422E87183572953916B46497F46A2B7\dls_bootstrap.lib ..\..\dscanner-0.5.5\dscanner\.dub\build\library-debug-windows-x86-dmd_2080-7B91882D2ED2F1543EED8F2E6793599C\dscanner.lib ..\..\dsymbol-0.3.7\dsymbol\.dub\build\library-debug-windows-x86-dmd_2080-65985B8189D883258F05B5158B1B27B9\dsymbol.lib ..\..\emsi_containers-0.8.0-alpha.7\emsi_containers\.dub\build\unittest-debug-windows-x86-dmd_2080-6D0F9B496921F1996DB6F8501908644C\emsi_containers.lib ..\..\inifiled-1.3.1\inifiled\.dub\build\library-quiet-debug-windows-x86-dmd_2080-374996D3823593854D28BB912F9FC9A7\inifiled.lib ..\..\libddoc-0.3.0-beta.1\libddoc\.dub\build\lib-debug-windows-x86-dmd_2080-FC7BAC09A42AA4849EDE9A05A7CD60FF\ddoc.lib ..\..\libdparse-0.8.6\libdparse\.dub\build\library-debug-windows-x86-dmd_2080-B8D9FE8B41F275021C55D503787491ED\dparse.lib ..\..\stdx-allocator-2.77.2\stdx-allocator\.dub\build\library-debug-windows-x86-dmd_2080-FBEF841F920B6EEFDEF7C0F95E63093D\stdx-allocator.lib ..\..\dub-1.9.0\dub\.dub\build\library-debug-windows-x86-dmd_2080-16985499C7FE665A730DEE765A5B6063\dub.lib curl.lib -g
```

![image](https://user-images.githubusercontent.com/18348637/40579299-b26a9fa6-6124-11e8-8078-69df4a2db2bf.png)

**Comment by LaurentTreguier** [comment]
Module names are supposed to match file names in D as far as I know. DLS is going to do the same as we use the same backend for completion: DCD.

To build it you need to use `--arch=x86_mscoff` or `--arch=x86_64`. But since you have the libcurl.dll problem, `dls.exe` is probably also going to crash anyway.

If you **really** want to try and make it work, I would advise you to [grab the binary release zip](https://github.com/LaurentTreguier/dls/releases/tag/v0.5.0), unpack it somewhere, put `libcurl.dll` manually next to `dls.exe` (a 32bit and a 64bit version can be found under `c:\d\dmd2\windows\bin` and `c:\d\dmd2\windows\bin64`, and then set `d.dlsPath` to the path to `dls.exe`. But at this point I'm afraid Windows can still give some trouble...

**Comment by RUSshy** [comment]
> Module names are supposed to match file names in D as far as I know

I meant issue this: https://github.com/Pure-D/code-d/issues/180

> DLS is going to do the same as we use the same backend for completion: DCD.

So you mean subPackages isn't supported too ?

**Comment by LaurentTreguier** [comment]
Subpackages managed by dub are supported. If you aren't using dub for the project, then the only solution would be to set `d.dls.symbol.importPaths` for this extension.

**Comment by RUSshy** [comment]
I am using dub, bu supported i mean if ``myapp`` reference ``mylib``, will ``myapp`` get code completion for types in ``mylib`` ?

**Comment by LaurentTreguier** [comment]
If `mylib` is set up in your `dub.json` or `dub.sdl` file as a dependency of `myapp`, then it should work

**Comment by LaurentTreguier** [comment]
Looks like there is a mismatch between the two versions as one is using a "v" and not the other... Must be a regression from changing the update system. This one shouldn't be too hard to fix.

**Comment by LaurentTreguier** [comment]
Version `0.5.1` should fix the update prompting problem (one of the next notifications you see should be a one with a real update 😄)

**Comment by LaurentTreguier** [comment]
Alright, so Windows looks into the `PATH` environment variable to search for libraries. Postgresql's bin directory is probably in it, which is why Windows attempts to use its `libcurl.dll`. Assuming you have dmd installed on your Windows system, you could make `dls:bootstrap` work by changing your `PATH` and putting dmd's bin path before Prostgresql's bin path. After `dub run dls:bootstrap`, you could set your `PATH` variable exactly like before, as I'm now shipping `libcurl.dll` along with `dls.exe`.

However, even this works, you'll still end up with `dls.exe` crashing because of bugs... But's that's another story altogether.

**Comment by LaurentTreguier** [comment]
This should be completely resolved with the release of dls 0.5.6.
By now, symlinks are created after asking for admin rights, `libcurl.dll` is bundled with `dls.exe` and `dls:bootstrap` auto-injects the path to DMD's libcurl into itself to use the correct dll (which might become the case for dls later on as well).
